dotnet-desktop-guide/net/winforms/wfdev-diagnostics/wfdev004.md

```diff
@@ -0,0 +1,61 @@
+---
+title: WFDEV004 warning
+description: Learn about Windows Forms compile-time warning WFDEV004. Form.OnClosing, Form.OnClosed, Form.Closing, and Form.Closed, are obsolete.
+ms.date: 03/04/2025
+f1_keywords:
+  - "WFDEV004"
+helpviewer_keywords:
+  - "WFDEV004"
+---
+# Compiler Warning WFDEV004
+
+**Version introduced:** .NET 10 Preview 1
+
+> [Form.OnClosing], [Form.OnClosed] and the corresponding events are obsolete. Use [Form.OnFormClosing], [Form.OnFormClosed], [Form.FormClosing] and [Form.FormClosed] instead.
+
+These methods and events are provided for backwards compatibility and they shouldn't be used. Instead, reference the replacement methods and events.
+
+## Workaround
+
+Replace the obsolete member with the new member:
+
+| Old member       | New member           |
+|------------------|----------------------|
+| [Form.OnClosing] | [Form.OnFormClosing] |
```

Comment:
In other places method names are enclosed in ticks, not square brackets, are these intentional?

Reply:
In markdown these match the link definitions at the end of the article. This is a way to create a link ID and then reuse it. Makes it easier to write in some cases.